plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -344,7 +350,17 @@ maybeAddSubst a s n rhs = do
             let termUsedAtMostOnce = Usages.getUsageCount n usgs <= 1
             -- See Note [Inlining and purity]
             termIsPure <- checkPurity t
-            pure $ termUsedAtMostOnce && case s of { Strict -> termIsPure; NonStrict -> True; }
+            -- This can in the worst case traverse a lot of the term, which could lead to us
+            -- doing ~quadratic work as we process the program. However in practice most term
```

Comment:
We don't have benchmarking for the plugin, do we? Either way, quadratic should be fine; I'd be much more worried if it's O(n^3) or higher.

Reply:
No, we have no compilation benchmarks. It would be nice to have. I think we could pull out the PIR that the plugin saves (I think it's the initial PIR) and then compile that in a benchmark. I'll make a ticket.

Reply:
https://input-output.atlassian.net/browse/PLT-582